src/main.rs

```diff
@@ -102,6 +105,12 @@ fn main() {
 				cache.window.win_x = new_pos.x;
 				cache.window.win_y = new_pos.y;
 			}
+			WindowEvent::KeyboardInput {
+				input: KeyboardInput { virtual_keycode: Some(VirtualKeyCode::Escape), .. },
+				..
+			} => {
+				bottom_container.set_visible(true);
+			}
```

Comment:
Setting the visibility of the bottom panel is tightly coupled to the logic that sets toggles the full screen mode. Therefore this should be together with that logic. I know it's not the nice but so far all Emulsion specific keyboard input is handled within the `picture_widget.rs` module. To avoid having a single logic spread out to distant functions, this should be handled within `picture_widget.rs` (at least for now). You will find there that toggle fullscreen mode appears twice with copied code. If you extracted into a function that would be great.